app/src/main/report/components/Artefacts/ArtefactsList.tsx

```diff
@@ -18,13 +20,15 @@ export class ArtefactsList extends React.Component<ArtefactsListProps, undefined
     }
 
     render() {
-
         const artefactItems = this.props.artefacts
-            .map((artefact)=> {
-                const key = Object.getOwnPropertyNames(artefact)[0];
-                return <ArtefactItem key={key} report={this.props.report} version={this.props.version} {...artefact[key]} />;
+            .map((artefactLookup)=> {
```

Comment:
I presume this is the bulk of the actual code change - at some point it would be good if someone could walk me a little through how this is all put together as this mostly shows me I have no idea how things are being done

Reply:
(that's not at all a criticism of the code! Just me not knowing how this is all put together)

Reply:
To be honest, this code is fairly hard to read, owing the unusual JSON format Orderly is using. Data in the format you suggested as alternative in the ticket would be a lot more normal:

```
[
    {
        "type": "data",
        "filenames": [
            "all.csv"
        ],
        "description": "raw export"
    },
    {
        "type": "data",
        "filenames": [
            "subset.csv"
        ],
        "description": "the subset we care most about"
    },
    ...
]
```

Then this could read something like:

```
const artefactItems = this.props.artefacts
            .map(artefact => {
                return <ArtefactItem key={artefact.description} report={this.props.report} version={this.props.version} {...artefact} />;
            });
```

Reply:
It's more that I don't have a model of what is going on in general (in a language I've never used).  But yes, that does look easier to read too.

One option that @littledice suggested was that we could move from storing json to working with relational data in the SQLite database.  That might be simpler than dealing with json strings.  I can discuss that with her when she's back perhaps